framework/test/unit/modules/dpos_v2/module.spec.ts

```diff
@@ -1282,6 +1035,202 @@ describe('DPoS module', () => {
+					stateStore,
+				}).getBlockAfterExecuteContext();
+				// Forger selection relies on vote weight to be sorted
+				const delegates: { address: Buffer; weight: bigint }[] = [
+					...scenario.testCases.input.voteWeights.map(d => ({
+						address: Buffer.from(d.address, 'hex'),
+						weight: BigInt(d.voteWeight),
+					})),
+				].slice(10);
+				sortValidatorsByWeightDesc(delegates);
+
+				// Overwrite the snapshot validator address to be the one in init delegates
+				const [duplicateAddress] = initDelegates;
+				delegates[0].address = duplicateAddress;
+
+				const result = await dpos['_getActiveDelegates'](context, delegates, 6);
+
+				expect(result).toHaveLength(defaultConfigs.numberActiveDelegates);
+			});
+		});
+
+		describe('when current round is more than initRounds + numberOfActiveDelegates', () => {
+			it('should all if snapshotValidators is less than numberOfActiveDelegates', async () => {
+				context = createBlockContext({
+					stateStore,
+				}).getBlockAfterExecuteContext();
+				// Forger selection relies on vote weight to be sorted
+				const delegates: { address: Buffer; weight: bigint }[] = [
+					...scenario.testCases.input.voteWeights.map(d => ({
+						address: Buffer.from(d.address, 'hex'),
+						weight: BigInt(d.voteWeight),
+					})),
+				].slice(0, 10);
+				sortValidatorsByWeightDesc(delegates);
+
+				const result = await dpos['_getActiveDelegates'](context, delegates, 104);
+				expect(result).toHaveLength(10);
+				expect(dpos['_capWeight']).not.toHaveBeenCalled();
+			});
+
+			it('should numberOfActiveDelegates if snapshotValidators is longer', async () => {
+				context = createBlockContext({
+					stateStore,
+				}).getBlockAfterExecuteContext();
+				// Forger selection relies on vote weight to be sorted
+				const delegates: { address: Buffer; weight: bigint }[] = [
+					...scenario.testCases.input.voteWeights.map(d => ({
+						address: Buffer.from(d.address, 'hex'),
+						weight: BigInt(d.voteWeight),
+					})),
+				];
+				sortValidatorsByWeightDesc(delegates);
+
+				const result = await dpos['_getActiveDelegates'](context, delegates, 104);
+				expect(result).toHaveLength(defaultConfigs.numberActiveDelegates);
+			});
+
+			it('should cap the weight if activeDelegates is more than capValue', async () => {
+				context = createBlockContext({
+					stateStore,
+				}).getBlockAfterExecuteContext();
+				// Forger selection relies on vote weight to be sorted
+				const delegates: { address: Buffer; weight: bigint }[] = [
+					...scenario.testCases.input.voteWeights.map(d => ({
+						address: Buffer.from(d.address, 'hex'),
+						weight: BigInt(d.voteWeight),
+					})),
+				];
+				sortValidatorsByWeightDesc(delegates);
+
+				await dpos['_getActiveDelegates'](context, delegates, 104);
+				expect(dpos['_capWeight']).toHaveBeenCalledWith(
+					expect.any(Array),
+					defaultConfigs.maxBFTWeightCap,
+				);
+			});
+		});
+	});
+
+	describe('_capWeight', () => {
```

Comment:
@gkoumout Could you check if this test case is correct?

Reply:
Looks good to me.